editor/editor_node.cpp

```diff
@@ -5372,8 +5377,16 @@ bool EditorNode::progress_task_step(const String &p_task, const String &p_state,
 	if (!singleton) {
 		return false;
 	} else if (singleton->cmdline_mode) {
-		const int percent = (p_step / float(progress_total_steps + 1)) * 100;
-		print_line_rich(vformat("[%4d%% ] [color=gray][b]%s[/b] | %s[/color]", percent, p_task, p_state));
+		double current_time = USEC_TO_SEC(OS::get_singleton()->get_ticks_usec());
+		double elapsed_time = current_time - last_progress_time;
+		if (last_progress_task != p_task || last_progress_state != p_state || last_progress_step != p_step || elapsed_time >= 1.0) {
+			const int percent = (p_step / float(progress_total_steps + 1)) * 100;
+			print_line_rich(vformat("[%4d%% ] [color=gray][b]%s[/b] | %s[/color]", percent, p_task, p_state));
+			last_progress_task = p_task;
+			last_progress_state = p_state;
+			last_progress_step = p_step;
+			last_progress_time = current_time;
+		}
```

Comment:
```suggestion
		if (last_progress_task != p_task || last_progress_state != p_state || last_progress_step != p_step || elapsed_time >= 1.0) {
			// Only print progress if it's changed since the last print.
			// This prevents multithreaded import from printing the same progress several times,
			// which would bloat log files.
			const int percent = (p_step / float(progress_total_steps + 1)) * 100;
			print_line_rich(vformat("[%4d%% ] [color=gray][b]%s[/b] | %s[/color]", percent, p_task, p_state));
			last_progress_task = p_task;
			last_progress_state = p_state;
			last_progress_step = p_step;
			last_progress_time = current_time;
		}
```